src/marqo/tensor_search/tensor_search.py

```
@@ -635,6 +640,152 @@ def gather_documents_from_response(response: QueryResult, marqo_index: MarqoInde
+    for group in root_groups:
+        if group.children is None:
+            continue
+        for field in group.children:
+            field_name = _extract_facet_field_name(field.id, facets)
+            facets_response.setdefault(field_name, {})
+
+            for value in field.children:
+                group_type, value_key = _parse_value_id(value.id)
+                processed_stats = _process_value_stats(value.fields)
+
+                if facets.fields[field_name].type in ["string", "array"]:
+                    if value.id == "group:string:":
+                        # Vespa's value for not found
+                        continue
+                    # values might be nested deeply if user data is stored as a.b.c
+                    current_level = facets_response[field_name]
+                    for key in value_key:
+                        if key not in current_level:
+                            current_level[key] = {}
+                        current_level = current_level[key]
+                    current_level.update(processed_stats)
+                else:
+                    if any(value in [-9223372036854775808, 'NaN'] for value in processed_stats.values()):
+                        # Vespa's value for null for int and float
+                        continue
+                    if facets.fields[field_name].ranges is None:
+                        if any(value in [-9223372036854775808, 'NaN'] for value in processed_stats.values()):
+                            # Vespa's value for null
+                            continue
+                        # aggregate statistic between int and float
+                        facets_response[field_name] = _combine_number_stats(
+                            facets_response.get(field_name, {}), processed_stats
+                        )
+                    elif group_type != "null": # the case when there is no values in range
```

What's the `else` here? If these are expected to be exhaustive, you can error out for `else` as an invariant violation (bug). This seems like it could fail silently

---

i.e. what happens if `ranges is None and group_type == null and type not in ["string", "array"]`

---

Moved comments to new location